extensions/_beacons/http_status.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+'''
+Beacon to manage and report the status of a server status endpoint.
+Fire an event when specified values don't match returned response.
+
+'''
+
+# Import Python libs
+from __future__ import absolute_import, unicode_literals
+import logging
+import operator
+import re
+import requests
+import salt.utils
+from salt.ext.six.moves import map
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'http_status'
+
+comparisons = {
+  '=': operator.eq,
+  '<': operator.lt,
+  '>': operator.gt,
+  '<=': operator.le,
+  '>=': operator.ge,
+  '!=': operator.ne,
+  'search': re.search
+}
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def validate(config):
+    '''
+    Validate the beacon configuration
+    '''
+    if not isinstance(config, list):
+        return False, ('Configuration for %s beacon must '
+                       'be a list.', __virtualname__)
+    else:
+        _config = {}
+        list(map(_config.update, config))
+
+        if 'sites' not in _config:
+            return False, ('Configuration for %s beacon '
+                           'requires sites.', __virtualname__)
+        else:
+            if not isinstance(_config['sites'], list):
+                return False, ('Sites for %s beacon '
+                               'must be a list.', __virtualname__)
+            else:
+                for sites in _config['sites']:
+                    log.debug('_config %s', _config['sites'][sites])
+                    if 'status_endpoint' not in _config['sites'][sites]:
```

Comment:
I would argue that `status_endpoint` is a bit too specific. I would set this to be something like `url` as it is more generic for the case where we are just checking that an HTTP(S) endpoint returns a 200